lang/compiler.go

```diff
@@ -106,7 +106,7 @@ func CompileAST(astPkg *ast.Package, now time.Time, opts ...CompileOption) *AstP
 		Program: &Program{
 			opts: applyOptions(opts...),
 		},
-		Ast: astPkg,
+		Ast: astPkg.Copy().(*ast.Package),
```

Comment:
Why do we need to copy here?  I would have thought that the AST is read-only from here.

Reply:
Your changes to the `AstProgram` means it can modify the package struct now.

Reply:
I modified this back so the copy happens lazily in the `Start` method only when it is modifying the package that way we aren't getting hit with a potentially unnecessary copy.